docs/metrics.md

```diff
@@ -58,67 +155,241 @@ at the current time, they should not be used to report historical data such as:
 a way to view and analyze historical data, consider the [workflow archive](workflow-archive.md) or reporting to logs.
 
+### Counter, gauge and histogram
+
+The following terms have specific meanings for the metrics that we collect:
+
+- `counter`: A value that accumulates over time – you can think of this like an odometer on a car; it only ever goes up.
+    - The rate of change of counters can be a very powerful tool in understanding these metrics.
+- `gauge`: Measures a current value at the time it is read. An example would be the fuel gauge in a vehicle.
+- `histogram`: A client-side aggregation of values, such as request latencies. A histogram is a good choice if you are interested in value statistics. For example: How many requests take fewer than 1 second?
+
+Thanks to [OpenTelemetry](https://opentelemetry.io/docs/concepts/signals/metrics/#metric-instruments) for these explanations.
+
 ### Default Controller Metrics
 
 Metrics for the Four Golden Signals are:
 
-- Latency: `argo_workflows_queue_latency`
-- Traffic: `argo_workflows_count` and `argo_workflows_queue_depth_count`
-- Errors: `argo_workflows_count` and `argo_workflows_error_count`
-- Saturation: `argo_workflows_workers_busy` and `argo_workflows_workflow_condition`
+- Latency: `queue_latency`
+- Traffic: `gauge` and `queue_depth_gauge`
+- Errors: `count` and `error_count`
+- Saturation: `workers_busy` and `workflow_condition`
 
+!!! High cardinality
+    Some metric attributes may have high cardinality and are marked with ⚠️ to warn you. You may need to disable this metric or disable the attribute.
 <!-- titles should be the exact metric name for deep-linking, alphabetical ordered -->
+<!-- titles are without argo_workflows prefix -->
+#### `build_info`
+
+The build information for this workflow controller
+
+| attribute   | explanation                                                            |
+|-------------|------------------------------------------------------------------------|
+| `version`   | The version of argo-workflows                                          |
+| `platform`  | Platform this is running on, as go describes it e.g. `linux/amd64` |
+| `gover`     | Version of go that built this workflow controller                  |
+| `build`     | Build date for this workflow controller                                |
+| `compiler`  | The compiler used to build this workflow controller e.g. `gc`          |
+| `commit`    | The full git SHA1 of the code that built this workflow controller      |
+| `treestate` | Whether the git tree was `dirty` or `clean` when this was built        |
+| `tag`       | The tag on the git commit or `untagged` if it was not tagged           |
+
+#### `cronworkflows_triggered_total`
+
+A counter of the number of times a CronWorkflow has been
+
+| attribute   | explanation                               |
+|-------------|-------------------------------------------|
+| `name`     | ⚠️ The name of the CronWorkflow. |
```

Comment:
I think it would make more sense to have the icon next to the attribute rather than its explanation -- much easier to see that way:

```suggestion
| ⚠️ `name`     | The name of the CronWorkflow. |
```

Also we should consider automatically generating all these from the go types as we do with the Fields Reference and others

Reply:
Automatically generating would be lovely, but more scope creep, so not doing it now.

Reply:
I disagree on the icon positioning, and prefer it with the description. It is in effect a concise part of the description, and I don't feel it is important enough to warrant moving left.

Reply:
Since you did write a "High Cardinality" warning already, it did sound important enough 😅